test/OpenTelemetry.Instrumentation.AspNetCore.Tests/MetricTests.cs

```diff
@@ -83,11 +91,134 @@ public async Task RequestMetricIsCaptured()
+        // In unit tests environment there may be a lot of parallel unit tests executed, so
+        // giving some breezing room for the End callback to complete
+        await Task.Delay(TimeSpan.FromSeconds(1)).ConfigureAwait(false);
+
+        this.meterProvider.Dispose();
+
+        var requestMetrics = metricItems
+            .Where(item => item.Name == "http.server.request.duration")
+            .ToArray();
+
+        var metric = Assert.Single(requestMetrics);
+
+        Assert.Equal("s", metric.Unit);
         var metricPoints = GetMetricPoints(metric);
         Assert.Equal(2, metricPoints.Count);
 
-        AssertMetricPoint(metricPoints[0], expectedRoute: "api/Values");
-        AssertMetricPoint(metricPoints[1], expectedRoute: "api/Values/{id}");
+        AssertMetricPoints_New(
+            metricPoints: metricPoints,
+            expectedRoutes: new List<string> { "api/Values", "api/Values/{id}" },
+            expectedTagsCount: 5);
+    }
+
+    [Fact]
+    public async Task RequestMetricIsCaptured_Dup()
+    {
+        var configuration = new ConfigurationBuilder()
+            .AddInMemoryCollection(new Dictionary<string, string> { [SemanticConventionOptInKeyName] = "http/dup" })
+            .Build();
+
+        var metricItems = new List<Metric>();
+
+        this.meterProvider = Sdk.CreateMeterProviderBuilder()
+            .ConfigureServices(services => services.AddSingleton<IConfiguration>(configuration))
+            .AddAspNetCoreInstrumentation()
+            .AddInMemoryExporter(metricItems)
+            .Build();
+
+        using (var client = this.factory
+            .WithWebHostBuilder(builder =>
+            {
+                builder.ConfigureLogging(loggingBuilder => loggingBuilder.ClearProviders());
+            })
+            .CreateClient())
+        {
+            using var response1 = await client.GetAsync("/api/values").ConfigureAwait(false);
+            using var response2 = await client.GetAsync("/api/values/2").ConfigureAwait(false);
+
+            response1.EnsureSuccessStatusCode();
+            response2.EnsureSuccessStatusCode();
+        }
+
+        // We need to let End callback execute as it is executed AFTER response was returned.
+        // In unit tests environment there may be a lot of parallel unit tests executed, so
+        // giving some breezing room for the End callback to complete
+        await Task.Delay(TimeSpan.FromSeconds(1)).ConfigureAwait(false);
+
+        this.meterProvider.Dispose();
+
+        // Validate Old Semantic Convention
+        var requestMetrics = metricItems
+            .Where(item => item.Name == "http.server.duration")
+            .ToArray();
+
+        var metric = Assert.Single(requestMetrics);
+        Assert.Equal("ms", metric.Unit);
+        var metricPoints = GetMetricPoints(metric);
+        Assert.Equal(2, metricPoints.Count);
+
+        AssertMetricPoints_Old(
```

Comment:
to simplify this, we should remove multiple route values as that is not what we are testing here. This can be done as a follow up as its not something that is introduced in this PR.